source/gateway.go

```diff
@@ -223,22 +215,28 @@ func (sc *gatewaySource) setResourceLabel(config istiomodel.Config, endpoints []
 	}
 }
 
-func (sc *gatewaySource) targetsFromIstioIngressGatewayServices() (targets endpoint.Targets, err error) {
-	for _, lbService := range sc.istioIngressGatewayServices {
-		lbNamespace, lbName, err := parseIngressGateway(lbService)
-		if err != nil {
-			return nil, err
-		}
-		if svc, err := sc.kubeClient.CoreV1().Services(lbNamespace).Get(lbName, metav1.GetOptions{}); err != nil {
-			log.Warn(err)
-		} else {
-			for _, lb := range svc.Status.LoadBalancer.Ingress {
-				if lb.IP != "" {
-					targets = append(targets, lb.IP)
-				}
-				if lb.Hostname != "" {
-					targets = append(targets, lb.Hostname)
-				}
+func (sc *gatewaySource) targetsFromGatewayConfig(config *istiomodel.Config) (targets endpoint.Targets, err error) {
+	gateway := config.Spec.(*istionetworking.Gateway)
+
+	listOptions := metav1.ListOptions{
+		LabelSelector: labels.Set(gateway.Selector).String(),
+	}
+
+	services, err := sc.kubeClient.CoreV1().Services(sc.namespace).List(listOptions)
```

Comment:
it would be better to use an shared informer which watches on such events instead of calling everytime, see PR: https://github.com/kubernetes-incubator/external-dns/pull/917

Reply:
Like this: https://github.com/kubernetes-incubator/external-dns/blob/master/source/service.go#L84 in NewIstioGatewaySource